package exception;

public class mainClass {

	public static void main(String[] args) {
		/*
		 
		 	Exception : 예외 != 에러 -> 개발자용 


			Baseball -> 1 ~ 10 -> 'A' == 65
			random find -> 1 ~ 100 -> 'B' == 66
			number format 10 -> 'a' == 97
			array index [5] -> [0] ~ [4] -> [5] 
			class not find
			file not found 

			형식
			
			try[
				// 예외가 발생할 가능성이 있는 코드
				  
			]catch(예외클래스1 e{
				 // 예외 메세지를 발생
			}catch(예외클래스2 e{
				 // 예외 메세지 발생2
			}...............3....
			}finally{
				// 무조건 실행
				// 되롤림(undo)
				// rollback 
			}
		 */

		int arr[] = {11, 22, 33 };
		
		System.out.println("프로그램 시작");
		
		try {
		
			for (int i = 0; i < 4; i++) {
				System.out.println(arr[i]);
			}
			System.out.println("for문 탈출 부분");
		}catch (ArrayIndexOutOfBoundsException e) {
			// System.out.println("배열의 범위를 벗어났습니다.");
			// e.printStackTrace();
			System.out.println(e.getMessage());
		}catch (NumberFormatException e) {
			
		}
	}

}
